### PR TITLE
[DOC-3283] Add replica pod deletion

### PR DESCRIPTION
**Pull request by brian-f-harness**
Add replica pod deletion

For reviewers, previews are available:

[Graceful delegate shutdown](https://64a3306357067b554593f247--harness-developer.netlify.app/docs/platform/Delegates/delegate-concepts/graceful-delegate-shutdown-process)
[Delete replica pods](https://649b53db45a7f00cebbac44d--harness-developer.netlify.app/docs/platform/Delegates/manage-delegates/delete-a-delegate#delete-replica-pods)

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [x] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/platform/2_Delegates/manage-delegates/delete-a-delegate.md [hunk 1]
```diff
@@ -61,6 +61,11 @@ For example, if the **Deployment** name is `quickstart-delegate`:
 Locate the delegate in the Harness account/Project/Org, click more options (⋮), and then click **Delete**.
 
 ![](static/delete-a-delegate-15.png)
+
+### Replica pod deletion
+
+Deleted replica pods unregister and clear out during shutdown after they complete all running tasks if the graceful shutdown period is sufficient, which is 10 minutes by default. During abnormal shutdown past the graceful shutdown period, replica pods clear out automatically in six hours. For more information on graceful shutdown, go to [Graceful delegate shutdown](/docs/platform/delegates/delegate-concepts/graceful-delegate-shutdown-process/).
```
Comment (xingchi-jin) [inline docs/platform/2_Delegates/manage-delegates/delete-a-delegate.md:67]: discussed in DM. the six hours is a TTL on harness ui. Thanks Brian for making the change

Comment (brian-f-harness) [reply docs/platform/2_Delegates/manage-delegates/delete-a-delegate.md]: Thank you for your review. Updated the content.

File: docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md [hunk 2]
```diff
@@ -14,17 +14,101 @@ Harness Delegate is designed to shut down gracefully. The process of graceful de
...
+         allowPrivilegeEscalation: false  
+         runAsUser: 0   
+```
+
+### ECS deployment default interval configuration
+
+Open the delegate manifest file and locate the container `containerDefinitions`. Change the `stopTimeout` as shown in the following YAML. In the example below, `stopTimeout` is set to 10 minutes.
+
+```yaml
+ipcMode:
```
Comment (xingchi-jin) [inline docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md:48]: the format/indent looks a bit off. we have another ecs task definition file https://developer.harness.io/docs/platform/delegates/install-delegates/docker-delegate-to-ecs-fargate/ 
could you make sure we use the same template file, so customer won't notice inconsistency between them

File: docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md [hunk 3]
```diff
@@ -14,17 +14,101 @@ Harness Delegate is designed to shut down gracefully. The process of graceful de
...
 
-| **Delegate environment** | **Trigger** |
-| :-- | :--: | :--:|
-| Kubernetes | Pod termination, eviction, or user-initiated scaling |
-| Docker | `docker kill` command |
-| Shell | `./stop.sh` instruction |
+| **Delegate environment** | **Trigger** 
+| :-- | :--: 
+| Kubernetes | Pod termination, eviction, or user-initiated scaling 
+| Docker | `docker kill` command
```
Comment (xingchi-jin) [inline docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md:113]: here should be 'docker stop'
docker stop gracefully terminates the container, docker kill forcefully kills the container

Comment (brian-f-harness) [reply docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md:113]: Updated

File: docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md (hunk 2 above)
Comment (brian-f-harness) [reply docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md:48]: @xingchi-jin I don't see stopTimeout in the ecs task definition file. Could you provide a sample that includes it or point me to where it should be added in the example?

File: docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md [hunk 4]
```diff
@@ -14,17 +14,114 @@ Harness Delegate is designed to shut down gracefully. The process of graceful de
...
+       imagePullPolicy: Always  
+       name: delegate  
+       securityContext:  
+         allowPrivilegeEscalation: false  
+         runAsUser: 0   
+```
+
+### Amazon ECS deployment default interval configuration
+
+Open the delegate manifest file and locate the container `containerDefinitions`. Change the `stopTimeout` as shown in the following JSON. In the example below, `stopTimeout` is set to 10 minutes.
```
Comment (xingchi-jin) [inline docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md:45]: the stopTimeout is missing in example

Comment (brian-f-harness) [reply docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md:45]: @xingchi-jin Yes, it's not included in the example. Can you help with where it should go in the file? I wasn't able to find example code with the stopTimeout included. Thanks for your help.

Comment (xingchi-jin) [reply docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md:45]: it's documented here https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definitions  - search for stopTimeout

within "containerDefinitions" , 

           "memory": 2048,
           "image": "harness/delegate:22.12.77802",
           "essential": true,
           "hostname": "<DELEGATE_HOST>",
           "name": "<DELEGATE_NAME>"
            "stopTimeout": ...

**Review by xingchi-jin — DISMISSED** [review]

File: docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md [hunk 5]
```diff
@@ -14,17 +14,119 @@ Harness Delegate is designed to shut down gracefully. The process of graceful de
...
+
 The length of the grace period is configurable. 
 
 | **Delegate type** | **Grace period** | **Default interval** |
 | :-- | :--: | :--: |
-| Immutable image | Yes | 10 minutes |
+| Immutable image | Yes | Configurable (details below)  |
 | Legacy image | No | 30 seconds |
 
+### Kubernetes deployment default interval configuration
```
Comment (StrawberryPad) [inline docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md:26]:
```suggestion
### Configure the default interval for a Kubernetes deployment
```

File: docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md [hunk 6]
```diff
@@ -14,17 +14,119 @@ Harness Delegate is designed to shut down gracefully. The process of graceful de
...
+     containers:  
+     - image: example/org:custom-delegate  
+       imagePullPolicy: Always  
+       name: delegate  
+       securityContext:  
+         allowPrivilegeEscalation: false  
+         runAsUser: 0   
+```
+
+### Amazon ECS deployment default interval configuration
```
Comment (StrawberryPad) [inline docs/platform/2_Delegates/delegate-concepts/graceful-delegate-shutdown-process.md:43]:
```suggestion
### Configure the default interval for an Amazon ECS deployment
```

**Review by StrawberryPad — APPROVED** [review]

**Pull request merged**